src/ui/widgets/XYPlot/xyPlot.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { CSSProperties } from "react";
+import { Widget } from "../widget";
+import {
+  InferWidgetProps,
+  ColorPropOpt,
+  StringPropOpt,
+  FontPropOpt,
+  TracesPropOpt,
+  AxesPropOpt,
+  BoolPropOpt,
+  FloatPropOpt
+} from "../propTypes";
+import { PVComponent, PVWidgetPropType } from "../widgetProps";
+import { registerWidget } from "../register";
+import Plot from "react-plotly.js";
+import {
+  calculateAxisLimits,
+  createAxes,
+  createTraces,
+  NewAxisSettings
+} from "./xyPlotOptions";
+import { Color } from "../../../types/color";
+
+export const XYPlotProps = {
+  height: FloatPropOpt,
+  width: FloatPropOpt,
+  plotBackgroundColor: ColorPropOpt,
+  title: StringPropOpt,
+  titleFont: FontPropOpt,
+  showLegend: BoolPropOpt,
+  showPlotBorder: BoolPropOpt,
+  showToolbar: BoolPropOpt,
+  traces: TracesPropOpt,
+  axes: AxesPropOpt
+};
+
+export type XYPlotComponentProps = InferWidgetProps<typeof XYPlotProps> &
+  PVComponent;
+
+export const XYPlotComponent = (props: XYPlotComponentProps): JSX.Element => {
+  const {
+    height = 250,
+    width = 400,
+    value,
+    plotBackgroundColor = Color.fromRgba(255, 255, 255),
+    title = "",
+    titleFont,
+    showLegend = true,
+    showPlotBorder,
+    // showToolbar, // TO DO - do we want a toolbar as well?
+    traces,
+    axes
+  } = props;
+  // TO DO - having all these checks is not ideal
+  if (
+    value?.value.arrayValue &&
+    axes &&
+    traces &&
+    titleFont &&
+    width &&
+    height
+  ) {
+    const bytesPerElement = value.value.arrayValue.BYTES_PER_ELEMENT;
+    // If data exists, creates traces to plot
+    const dataSet = createTraces(traces, value, bytesPerElement);
+    // Set up style
+    let style: CSSProperties = {};
+    if (showPlotBorder) {
+      style = { border: "1px solid black", padding: "1px" };
+    }
+    const font = titleFont?.css();
+    // Sometimes font is a string with "px" on the end....
```

Comment:
We do have to do this in other place. For example. the height and width are also sent with "px": https://github.com/dls-controls/cs-web-lib/blob/3453939456917a6e2a1fbafed9042dc9045ebfa6/src/ui/widgets/EmbeddedDisplay/embeddedDisplay.tsx#L65
So it might be worth turning this into a standalone function that can be re-used?